### PR TITLE
[Seedless Onboarding]: Settings password form

### LINKED ISSUE #2519: Signer Account settings page

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #2519

## How this PR fixes it

- Updates the design for the password form
- Adds a password strength check
- Uses `useWallet` on the settings page to determine display logic
- Dispatches events on success/fail when submitting the form

## ToDos

- [ ] - Extract strings
- [x] Update settings page automatically when logging in with google
- [x] Dispatch notifications on success/fail

## How to test it

TBD

## Screenshots

Screenshot 2023-10-17 at 11 28 35
Screenshot 2023-10-17 at 11 28 49
Screenshot 2023-10-17 at 11 29 00
Screenshot 2023-10-17 at 11 29 11

## Checklist
* [x] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/settings/SignerAccountMFA/PasswordForm.tsx [hunk 1]
```diff
@@ -17,7 +36,23 @@ type PasswordFormData = {
...
+// At least 12 characters, one lowercase, one uppercase, one number, one symbol
+const strongPassword = new RegExp('(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{12,})')
+// At least 9 characters, one lowercase, one uppercase, one number, one symbol
+const mediumPassword = new RegExp('((?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{9,}))')
+
+export const PasswordForm = () => {
+  const dispatch = useAppDispatch()
+  const mpcCoreKit = useMPC()
+  const [passwordStrength, setPasswordStrength] = useState<PasswordStrength>(PasswordStrength.weak)
+  const [passwordsMatch, setPasswordsMatch] = useState<boolean>(false)
```
Comment (schmanu) [inline src/components/settings/SignerAccountMFA/PasswordForm.tsx:54]: Should we move this to the react hook form validation?

File: src/components/settings/SignerAccountMFA/PasswordForm.tsx [hunk 2]
```diff
@@ -27,80 +62,187 @@ export const PasswordForm = ({ mpcCoreKit }: { mpcCoreKit: Web3AuthMPCCoreKit })
...
+                      display="flex"
+                      alignItems="center"
+                      gap={1}
+                      mt={1}
+                      className={
+                        passwordStrength === PasswordStrength.strong
+                          ? css.strongPassword
+                          : passwordStrength === PasswordStrength.medium
+                          ? css.mediumPassword
+                          : css.weakPassword
```
Comment (schmanu) [inline src/components/settings/SignerAccountMFA/PasswordForm.tsx:161]: We have two big condition blocks like this: 
For the css class and for the label, should we add a `Record<PasswordStrength, { label: string, cssClass: string}>` mapping?

File: src/components/settings/SignerAccountMFA/index.tsx [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 import { PasswordForm } from './PasswordForm'
+import useWallet from '@/hooks/wallets/useWallet'
+import { ONBOARD_MPC_MODULE_LABEL } from '@/services/mpc/module'
 
 const SignerAccountMFA = () => {
-  const mpcCoreKit = useMPC()
+  const wallet = useWallet()
 
-  if (mpcCoreKit?.status !== COREKIT_STATUS.LOGGED_IN) {
+  if (wallet?.label !== ONBOARD_MPC_MODULE_LABEL) {
```
Comment (schmanu) [inline src/components/settings/SignerAccountMFA/index.tsx:10]: Nit: I think we are using this check quite frequently across the app.
Should we add a small helper `isSocialLogin(wallet)`?

File: src/components/settings/SignerAccountMFA/PasswordForm.tsx [hunk 4]
```diff
@@ -17,7 +36,23 @@ type PasswordFormData = {
...
 
-export const PasswordForm = ({ mpcCoreKit }: { mpcCoreKit: Web3AuthMPCCoreKit }) => {
+enum PasswordStrength {
+  strong,
+  medium,
+  weak,
+}
+
+// At least 12 characters, one lowercase, one uppercase, one number, one symbol
+const strongPassword = new RegExp('(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{12,})')
```
[the remark this answers is not in the document]
Comment (usame-algan) [reply src/components/settings/SignerAccountMFA/PasswordForm.tsx:46]: I extracted the logic into a helper function and wrote a few tests for it.

File: src/components/settings/SignerAccountMFA/index.tsx (hunk 3 above)
Comment (usame-algan) [reply src/components/settings/SignerAccountMFA/index.tsx]: done

**Review by schmanu — COMMENTED** [review]
The form looks great now!

I have a small suggestion towards how to get rid of the ts-ignore annotations in an elegant way.

File: src/components/settings/SignerAccountMFA/helper.ts [hunk 5]
```diff
@@ -43,8 +46,24 @@ export const enableMFA = async (
...
+        groupKey: 'global-upsert-password',
+        message: 'Successfully created or updated password',
+      }),
+    )
   } catch (e) {
     const error = asError(e)
     logError(ErrorCodes._304, error.message)
+
+    dispatch(
+      showNotification({
```
[the remark this answers is not in the document]
Comment (usame-algan) [reply src/components/settings/SignerAccountMFA/helper.ts:63]: I added a comment in Figma to check with Design

**Review by schmanu — APPROVED** [review]
Vamos! 🚀

**Pull request merged**